Sources/Leaf/Stem+Spawn.swift

```diff
@@ -19,27 +19,31 @@ extension Stem {
         if name.hasPrefix("/") {
             name = String(name.characters.dropFirst())
         }
-
+        return try spawnLeaf(at: workingDirectory + name)
+    }
+    
+    public func spawnLeaf(at path: String) throws -> Leaf {
+        if let existing = cache?[path] {
+            return existing
+        }
+        
+        let leaf: Leaf
+        
         // non-leaf document. rendered as pure bytes
-        if name.characters.contains("."), !name.hasSuffix(".leaf") {
-            if let existing = cache?[name] { return existing }
-            let path = workingDirectory + name
+        if path.components(separatedBy: "/").last?.contains(".") == true, !path.hasSuffix(".leaf") {
```

Comment:
`foo` and `foo.leaf` should be rendered as leaf where as `foo.bar` should not.

the code got a little more complicated since the paths can have `path/../to/dir/foo.bar`